metis-indexing/metis-indexing-service/src/main/java/eu/europeana/indexing/service/PublishingService.java

```diff
@@ -0,0 +1,390 @@
+import eu.europeana.corelib.edm.utils.MongoConstructor;
+import eu.europeana.corelib.edm.utils.construct.SolrDocumentHandler;
+import eu.europeana.corelib.edm.utils.construct.TimespanUpdater;
+import eu.europeana.corelib.mongo.server.EdmMongoServer;
+import eu.europeana.corelib.mongo.server.impl.EdmMongoServerImpl;
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.corelib.solr.entity.AgentImpl;
+import eu.europeana.corelib.solr.entity.AggregationImpl;
+import eu.europeana.corelib.solr.entity.ConceptImpl;
+import eu.europeana.corelib.solr.entity.EuropeanaAggregationImpl;
+import eu.europeana.corelib.solr.entity.LicenseImpl;
+import eu.europeana.corelib.solr.entity.PlaceImpl;
+import eu.europeana.corelib.solr.entity.ProvidedCHOImpl;
+import eu.europeana.corelib.solr.entity.ProxyImpl;
+import eu.europeana.corelib.solr.entity.ServiceImpl;
+import eu.europeana.corelib.solr.entity.TimespanImpl;
+import eu.europeana.indexing.model.IndexingMongoServer;
+import eu.europeana.indexing.service.dao.FullBeanDao;
+import eu.europeana.metis.exception.IndexingException;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import org.mongodb.morphia.query.Query;
+import org.mongodb.morphia.query.UpdateOperations;
+import org.apache.commons.io.IOUtils;
+import org.apache.solr.client.solrj.impl.CloudSolrServer;
+import org.apache.solr.client.solrj.impl.LBHttpSolrServer;
+import org.apache.solr.client.solrj.SolrServer;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.SolrInputDocument;
+import org.jibx.runtime.BindingDirectory;
+import org.jibx.runtime.IBindingFactory;
+import org.jibx.runtime.IMarshallingContext;
+import org.jibx.runtime.IUnmarshallingContext;
+import org.jibx.runtime.JiBXException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PublishingService {
+	private static final Logger LOGGER = LoggerFactory.getLogger(PublishingService.class);
+	
+	private FullBeanDao fullBeanDao;	
+    private static IBindingFactory rdfBindingFactory;
+    private static final String UTF8 = StandardCharsets.UTF_8.name();
+    private SolrServer solrServer; 
+    private SolrServer cloudSolrServer;
+
+    static {    	
+    	try {
+    		rdfBindingFactory = BindingDirectory.getFactory(RDF.class);
+    	} catch (JiBXException e) {
+    		LOGGER.warn("Error creating the JibX factory.");    		
+    	}
+    }
+
+    public PublishingService(FullBeanDao fullBeanDao, LBHttpSolrServer solrServer, CloudSolrServer cloudSolrServer) {
+  	  this.fullBeanDao = fullBeanDao;
+  	  this.solrServer = solrServer;
+  	  this.cloudSolrServer = cloudSolrServer;
+    }
+    
+    public boolean process(String record) throws IndexingException
+    {    	

```

Comment:
Method is too complex (according to SonarQube... complexity of 16 instead of 15! ;)